### PR TITLE
fix error while creating new instance

### PR DESCRIPTION
**Pull request by LionelColaso**
fix new instance creation
image

### CODE REVIEW

File: app/views/main_window.py [hunk 1]
```diff
@@ -569,7 +572,7 @@ def clone_essential_paths(
             )
         )
         # Sanitize the input so that it does not produce any KeyError down the road
-        new_instance_name = self.__ask_for_new_instance_name().strip()
```
Comment (hk21702) [inline app/views/main_window.py:572]: You shouldn't be just removing the .strip(), then you're not sanitizing it

Fix the underlying issue.

File: app/views/main_window.py [hunk 2]
```diff
@@ -202,22 +202,25 @@ def initialize_content(self, is_initial: bool = True) -> None:
...
+            return instance_name.strip()
+        else:
+            return None
 
     def __ask_for_non_default_instance_name(self) -> str:
         while True:
-            instance_name, cancelled = show_dialogue_input(
+            instance_name, ok = show_dialogue_input(
                 title="Provide instance name",
                 text='Input a unique name for the backed up instance that is not "Default"',
```
Comment (hk21702) [inline app/views/main_window.py:218]: Change to label

File: app/views/main_window.py (hunk 1 above)
Comment (LionelColaso) [reply app/views/main_window.py:572]: it is 
def __ask_for_new_instance_name(self) -> str:
        instance_name, ok = show_dialogue_input(
            title="Create new instance",
            text="Input a unique name of new instance that is not already used:",
        )
        if ok:
            return instance_name.strip()
        else:
            return None

File: app/views/main_window.py [hunk 3]
```diff
@@ -202,29 +202,32 @@ def initialize_content(self, is_initial: bool = True) -> None:
             self.initialize_watchdog()
 
     def __ask_for_new_instance_name(self) -> str:
```
Comment (hk21702) [inline app/views/main_window.py:204]: return type should be str | None

File: app/views/main_window.py [hunk 4]
```diff
@@ -202,29 +202,32 @@ def initialize_content(self, is_initial: bool = True) -> None:
...
-            text="Input a unique name of new instance that is not already used:",
+            label="Input a unique name of new instance that is not already used:",
         )
-        return instance_name if not cancelled else None
+        if ok:
+            return instance_name.strip()
+        else:
+            return None
 
     def __ask_for_non_default_instance_name(self) -> str:
```
Comment (hk21702) [inline app/views/main_window.py:214]: return type should be str | None

**Comment by hk21702** [comment]
@twstagg can you take a look since you're more familiar with how instancing is supposed to work?

File: app/views/main_window.py [hunk 5]
```diff
@@ -201,23 +201,26 @@ def initialize_content(self, is_initial: bool = True) -> None:
...
-    def __ask_for_new_instance_name(self) -> str:
-        instance_name, cancelled = show_dialogue_input(
+    def __ask_for_new_instance_name(self) -> str | None:
+        instance_name, ok = show_dialogue_input(
             title="Create new instance",
-            text="Input a unique name of new instance that is not already used:",
+            label="Input a unique name of new instance that is not already used:",
         )
-        return instance_name if not cancelled else None
+        if ok:
```
Comment (twstagg) [inline app/views/main_window.py:209]: This could still be a single line.

**Review by twstagg — CHANGES_REQUESTED** [review]
Everything seems to work fine. i see that the strip is still happening. Looks good other than my nitpick.

**Review by twstagg — APPROVED** [review]
Thanks!

**Pull request merged**